ztp/tools/pgt2acm/main.go

```diff
@@ -0,0 +1,359 @@
+	}
+
+	var seenPoliciesSorted []string
+	for policyName := range seenPoliciesMap {
+		seenPoliciesSorted = append(seenPoliciesSorted, policyName)
+	}
+
+	sort.Strings(seenPoliciesSorted)
+	for _, policyName := range seenPoliciesSorted {
+		newPolicy := convertPGTPolicyToACMGenPolicy(&policyGenTemp, rootName, policyName, outputDir)
+		acmGenTempConversion.Policies = append(acmGenTempConversion.Policies, newPolicy)
+		var labelSelector map[string]interface{}
+		labelSelector, err = labels.OutputGeneric(labels.LabelToSelector(policyGenTemp.Spec.BindingRules,
+			policyGenTemp.Spec.BindingExcludedRules))
+		if err != nil {
+			return err
+		}
+
+		// Convert Miscelanous fields
+		convertSimpleMiscellaneousFields(&policyGenTemp, &acmGenTempConversion, rootName)
+
+		if workaroundPlacement != nil && !*workaroundPlacement {
+			acmGenTempConversion.PolicyDefaults.Placement.LabelSelector = labelSelector
+		} else {
+			// starts creating child policies as soon as the managed cluster starts installing
+			var placementFilepathRelative string
+			placementFilepathRelative, err = placement.GeneratePlacementFile(newPolicy.Name, acmGenTempConversion.PolicyDefaults.Namespace, outputDir, labelSelector)
+			if err != nil {
+				return fmt.Errorf("error when generating placement file, err: %s", err)
+			}
+			acmGenTempConversion.PolicyDefaults.Placement.PlacementPath = placementFilepathRelative
+		}
+
+		// Apply patches on ACMGen since it is not yet supported officially
+		if len(acmGenTempConversion.Policies) > 0 {
+			for policyIndex := range acmGenTempConversion.Policies {
+				for manifestIndex := range acmGenTempConversion.Policies[policyIndex].Manifests {
+					err = RenderPatchesInManifestForSpecifiedKinds(&policyGenTemp, &acmGenTempConversion, policyIndex, manifestIndex, outputDir, schema, preRenderPatchKindList)
+					if err != nil {
+						return fmt.Errorf("could not render patches in manifest, err:%s", err)
+					}
+				}
+			}
+		}
+	}
+	return writeConvertedTemplateToFile(&policyGenTemp, &acmGenTempConversion, outputFile)
+}
+
+func writeConvertedTemplateToFile(policyGenTemp *pgtformat.PolicyGenTemplate, acmGenTempConversion *acmformat.AcmGenTemplate, outputFile string) (err error) {
+	convertedContent, err := yaml.Marshal(acmGenTempConversion)
+	if err != nil {
+		return fmt.Errorf("could not marshall acm profile, err: %s", err)
+	}
+
+	convertedContent = []byte("---\n" + string(convertedContent))
+	convertedContent = []byte(strings.ReplaceAll(string(convertedContent), "$mcp", policyGenTemp.Spec.Mcp))
+
+	// Ensure the directory exists
+	dir := filepath.Dir(outputFile)
+	err = os.MkdirAll(dir, fileutils.DefaultDirWritePermissions)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(outputFile, convertedContent, fileutils.DefaultFileWritePermissions)
+	if err != nil {
+		fmt.Printf("Error writing to file, err: %s", err)
+		return err
+	}
+	fmt.Printf("Wrote converted ACM template: %s\n", outputFile)
```

Comment:
Do we need this printf comment, can't we simply do?
`fmt.Println(outputFile)`

Reply:
Yes, we are not just printing the file name, there is a prefix string.